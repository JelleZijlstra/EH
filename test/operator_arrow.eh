#!/usr/bin/ehi
include '../lib/tuple.eh'

# Test that we reject invalid index values
trier = func: f
	try
		f()
	catch
		echo exception
	end
end
trier func:
	[]->1.0 = 3
end
trier () => (printvar []->(1..3))
trier func:
	{}->1 = 3
end
trier () => (printvar {}->4)
trier func:
	(1, 2)->'test' = 42
end
trier () => ((1, 2)->'test')
