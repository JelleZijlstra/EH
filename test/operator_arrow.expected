%%stdout%%
<TypeError: Invalid type for argument to Array.operator->= (expected Integer): Float>
<TypeError: Invalid type for argument to Array.operator-> (expected Integer): Range>
<TypeError: Invalid type for argument to Hash.operator->= (expected String): Integer>
<TypeError: Invalid type for argument to Hash.operator-> (expected String): Integer>
<NameError: Unknown member operator->= in object of type Tuple: (1, 2)>
<TypeError: Invalid type for argument to Tuple.operator-> (expected Integer): String>
%%stderr%%
